scripts/debug_tools/failure_lib.py

```diff
@@ -0,0 +1,47 @@
+import json
+import os
+
+
+def findPath(string, path_begin):
+    """ Find one path in string.
+    @return: the pair of the begin and end pos
+    """
+    path_end = string.find(' ', path_begin)
+    if path_end == -1:
+        path_end = len(string)
+    path = string[path_begin:path_end]
+    return path, path_end
+
+
+def changePaths(string, pathModifierFun):
+    """ Scan through the string and possibly replace all found paths.
+    @return: the modified string
+    """
+    result = ''
+    i = 0
+    while i < len(string):
+        # Everything is a path wich starts with a '/' and there is a whitespace
+        # after that
+        if string[i] == '/':
+            path, path_end = findPath(string, i)
+            path = pathModifierFun(path)
+            result += path
+            i = path_end - 1
+        else:
+            result += string[i]
```

Comment:
If there will be a lot of items in the `result` using a `list` and `join` before the return can be more performant.

Reply:
Agreed. String building seems really expensive here.